Spack recipe failed to build with gcc 10 (and 4.9)

On quartz I cloned spack master and configured it to use `gcc/10.2`. Building flux-sched@0.15 failed with

```
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'
6 errors found in build log:
     720                     from /var/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-uo2zab2mblvjrxb7br7o5dnqloj5cg7j/spack-src/resource/utilities/resource-query.cpp:39:
     721    /g/g92/churavy1/spack/opt/spack/linux-rhel7-broadwell/gcc-10.2.1/boost-1.76.0-aysbkz6dx3n5k4kdpsgiycsnzbvw7wxw/include/boost/detail/iterator.hpp:13:1: note: '#pragma message: This header is deprecated. Use <iterator> instead.
            '
     722       13 | BOOST_HEADER_DEPRECATED("<iterator>")
     723          | ^~~~~~~~~~~~~~~~~~~~~~~
     724      CXXLD    grug2dot
     725    /var/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-uo2zab2mblvjrxb7br7o5dnqloj5cg7j/spack-src/resource/utilities/resource-query.cpp: In function 'std::shared_ptr<Flux::resource_model::resource_context_t> init_resourc
            e_query(int, char**)':
  >> 726    /var/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-uo2zab2mblvjrxb7br7o5dnqloj5cg7j/spack-src/resource/utilities/resource-query.cpp:721:21: error: 'DBL_MAX' was not declared in this scope
     727      721 |     ctx->perf.min = DBL_MAX;
     728          |                     ^~~~~~~
     729    /var/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-uo2zab2mblvjrxb7br7o5dnqloj5cg7j/spack-src/resource/utilities/resource-query.cpp:46:1: note: 'DBL_MAX' is defined in header '<cfloat>'; did you forget to '#include <
            cfloat>'?
     730       45 | #include "config.h"
     731      +++ |+#include <cfloat>
     732       46 | #endif
  >> 733    make[4]: *** [resource_query-resource-query.o] Error 1
     734    make[4]: Leaving directory `/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-uo2zab2mblvjrxb7br7o5dnqloj5cg7j/spack-src/spack-build/resource/utilities'
  >> 735    make[3]: *** [all-recursive] Error 1
     736    make[3]: Leaving directory `/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-uo2zab2mblvjrxb7br7o5dnqloj5cg7j/spack-src/spack-build/resource/utilities'
  >> 737    make[2]: *** [all-recursive] Error 1
     738    make[2]: Leaving directory `/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-uo2zab2mblvjrxb7br7o5dnqloj5cg7j/spack-src/spack-build/resource'
  >> 739    make[1]: *** [all-recursive] Error 1
     740    make[1]: Leaving directory `/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-uo2zab2mblvjrxb7br7o5dnqloj5cg7j/spack-src/spack-build'
  >> 741    make: *** [all] Error 2
See build log for details:
  /var/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-uo2zab2mblvjrxb7br7o5dnqloj5cg7j/spack-build-out.txt
==> Error: flux-sched-0.15.0-uo2zab2mblvjrxb7br7o5dnqloj5cg7j: ProcessError: Command exited with status 2:
    'make' '-j16'
==> Error: flux-sched-0.15.0-uo2zab2mblvjrxb7br7o5dnqloj5cg7j: Package was not installed
==> Updating view at /g/g92/churavy1/src/DaggerExamples/Flux/.spack-env/view
==> Error: Installation request failed.  Refer to reported errors for failing package(s).
```

Same issue with `4.9.3`

```
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

6 errors found in build log:
     583                     from /var/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-hnjqdbccv5dvyyggqbit45iygeqdtm7o/spack-src/resource/utilities/resource-query.cpp:39:
     584    /g/g92/churavy1/spack/opt/spack/linux-rhel7-broadwell/gcc-4.9.3/boost-1.76.0-azc7qknkn2yy2xzt3gnqbzcqq5mt7bzj/include/boost/detail/iterator.hpp:13:67: note: #pragma message: This header is deprecated. Use <iterator> instead.
     585     BOOST_HEADER_DEPRECATED("<iterator>")
     586                                                                       ^
     587      CXXLD    grug2dot
     588    /var/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-hnjqdbccv5dvyyggqbit45iygeqdtm7o/spack-src/resource/utilities/resource-query.cpp: In function 'std::shared_ptr<Flux::resource_model::resource_context_t> init_resourc
            e_query(int, char**)':
  >> 589    /var/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-hnjqdbccv5dvyyggqbit45iygeqdtm7o/spack-src/resource/utilities/resource-query.cpp:721:21: error: 'DBL_MAX' was not declared in this scope
     590         ctx->perf.min = DBL_MAX;
     591                         ^
  >> 592    make[4]: *** [resource_query-resource-query.o] Error 1
     593    make[4]: Leaving directory `/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-hnjqdbccv5dvyyggqbit45iygeqdtm7o/spack-src/spack-build/resource/utilities'
  >> 594    make[3]: *** [all-recursive] Error 1
     595    make[3]: Leaving directory `/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-hnjqdbccv5dvyyggqbit45iygeqdtm7o/spack-src/spack-build/resource/utilities'
  >> 596    make[2]: *** [all-recursive] Error 1
     597    make[2]: Leaving directory `/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-hnjqdbccv5dvyyggqbit45iygeqdtm7o/spack-src/spack-build/resource'
  >> 598    make[1]: *** [all-recursive] Error 1
     599    make[1]: Leaving directory `/tmp/churavy1/spack-stage/spack-stage-flux-sched-0.15.0-hnjqdbccv5dvyyggqbit45iygeqdtm7o/spack-src/spack-build'
  >> 600    make: *** [all] Error 2

```

--- a/resource/modules/resource_match.cpp
+++ b/resource/modules/resource_match.cpp
@@ -23,6 +23,7 @@
 \*****************************************************************************/
 
 #include <cstdint>
+#include <limits>
 #include <sstream>
 #include <cerrno>
 #include <map>
@@ -344,7 +345,7 @@ static std::shared_ptr<resource_ctx_t> getctx (flux_t *h)
         set_default_args (ctx->args);
         ctx->perf.load = 0.0f;
         ctx->perf.njobs = 0;
-        ctx->perf.min = DBL_MAX;
+        ctx->perf.min = std::numeric_limits<double>::max();
         ctx->perf.max = 0.0f;
         ctx->perf.accum = 0.0f;
         ctx->matcher = nullptr; /* Cannot be allocated at this point */

--- a/resource/schema/infra_data.cpp
+++ b/resource/schema/infra_data.cpp
@@ -20,6 +20,7 @@
  *  See also:  http://www.gnu.org/licenses/
 \*****************************************************************************/
 
+#include <limits>
 #include "resource/schema/infra_data.hpp"
 
 namespace Flux {

--- a/resource/utilities/resource-query.cpp
+++ b/resource/utilities/resource-query.cpp
@@ -29,7 +29,6 @@
 #include <sstream>
 #include <algorithm>
 #include <iterator>
-#include <cstdint>
 #include <limits>
 #include <memory>
 #include <readline/readline.h>
@@ -719,7 +718,7 @@ static std::shared_ptr<resource_context_t> init_resource_query (int c,
 
     set_default_params (ctx);
     process_args (ctx, c, v);
-    ctx->perf.min = DBL_MAX;
+    ctx->perf.min = std::numeric_limits<double>::max();
     ctx->perf.max = 0.0f;
     ctx->perf.accum = 0.0f;
     if ( !(ctx->matcher = create_match_cb (ctx->params.matcher_policy))) {

--- a/src/test/docker/fedora33/Dockerfile
+++ b/src/test/docker/fedora33/Dockerfile
@@ -1,0 +1,33 @@
+FROM fluxrm/flux-core:fedora33
+
+LABEL maintainer="Stephen Herbein <herbein1@llnl.gov>"
+
+ARG USER=flux
+ARG UID=1000
+
+#  Enable PowerTools for development packages
+RUN sudo yum -y update \
+ && sudo yum -y install \
+	boost-devel \
+	boost-graph \
+	boost-system \
+	boost-filesystem \
+	boost-regex \
+	libxml2-devel \
+	readline-devel \
+	python3-pyyaml \
+	yaml-cpp-devel \
+ && sudo yum clean all
+
+# Add configured user to image with sudo access:
+#
+RUN \
+ if test "$USER" != "flux"; then  \
+      sudo groupadd -g $UID $USER \
+   && sudo useradd -g $USER -u $UID -d /home/$USER -m $USER \
+   && sudo sh -c "printf \"$USER ALL= NOPASSWD: ALL\\n\" >> /etc/sudoers" \
+   && sudo usermod -G wheel $USER; \
+ fi
+
+USER $USER
+WORKDIR /home/$USER

--- a/src/test/generate-matrix.py
+++ b/src/test/generate-matrix.py
@@ -159,4 +159,11 @@ matrix.add_build(
     docker_tag=True,
 )
 
+# Fedora33
+matrix.add_build(
+    name="fedora33",
+    image="fedora33",
+    docker_tag=True,
+)
+
 print(matrix)

